integration_test/warehouse/integration_test.go

```diff
@@ -1677,14 +1669,12 @@ func TestUploads(t *testing.T) {
 				{A: "namespace", B: namespace},
 				{A: "status", B: exportedData},
 			}...)
-			requireDownstreamEventsCount(t, ctx, db, fmt.Sprintf("%s.%s", namespace, "tracks"), events)
+			requireDownstreamEventsCount(t, ctx, db, fmt.Sprintf("%s.%s", namespace, "tracks"), events*2)
```

Comment:
@achettyiitr how come we had this test asserting the events count should have been `events`? Shouldn't have it been `events*2` considering that `allowMerge` is `false` and we're doing two syncs?

Same question for the 2 tests below i.e.
* `allowMerge=false,preferAppend=true,IsReplaySource=true`
* `allowMerge=false,preferAppend=true,sourceCategory=cloud"`